tests/component/component.cpp

```diff
@@ -45,6 +45,168 @@ TEST(Component, unset_name) {
     EXPECT_EQ(e, a);
 }
 
+TEST(Component, count) {
+    libcellml::Component parent;
+    parent.setName("parent");
+    libcellml::Component child1;
+    child1.setName("child1");
+    libcellml::Component child2;
+    child2.setName("child2");
+    libcellml::Component child3;
+    child3.setName("child3");
+    libcellml::Component child4;
+    child4.setName("child4");
+
+    EXPECT_EQ(0, parent.componentCount());
+
+    parent.addComponent(child1);
+    parent.addComponent(child2);
+    parent.addComponent(child3);
+    parent.addComponent(child4);
+    EXPECT_EQ(4, parent.componentCount());
+
+    child3.addComponent(child4);
+    parent.addComponent(child3);
+    EXPECT_EQ(5, parent.componentCount());
+
+    EXPECT_EQ(1, child3.componentCount());
+}
+
+TEST(Component, contains) {
+    libcellml::Component c;
+    libcellml::Component c1, c2;
+    c1.setName("child1");
+    c2.setName("child2");
+
+    EXPECT_FALSE(c.containsComponent("child1"));
+
+    c.addComponent(c1);
+    c.addComponent(c2);
+    EXPECT_TRUE(c.containsComponent("child2"));
```

Comment:
Might also want to add:

```
EXPECT_TRUE(c.containsComponent("child1"));
```

or drop ```c2``` and do everything with ```c1```. (The same goes for ```TEST(Model, contains)```.)